src/edu/umass/ckc/wo/ttmain/ttcontroller/TeacherToolsReportController.java

```diff
@@ -34,37 +39,52 @@
 
 
     @RequestMapping(value = "/tt/getTeacherReports", method = RequestMethod.POST)
-    public @ResponseBody String getTeacherReport(ModelMap map, @RequestParam("teacherId") String teacherId, @RequestParam("classId") String classId, @RequestParam("reportType") String reportType,  @RequestParam("lang") String lang) throws TTCustomException {
-    	System.out.println("TeacherToolsReportController");
-    	return reportService.generateTeacherReport(teacherId, classId, reportType, lang);
+    public @ResponseBody String getTeacherReport(ModelMap map, @RequestParam("teacherId") String teacherId, @RequestParam("classId") String classId, @RequestParam("reportType") String reportType,  @RequestParam("lang") String lang,  @RequestParam("filter") String filter, HttpServletRequest request) throws TTCustomException {
+    	//  	System.out.println("Request = " + request.toString());
+    	//  	System.out.println("Request = " + request.getSession().getAttribute("conn").toString());
+    	  //  	int intId = (int) request.getSession().getAttribute("teacherId");
+    	  //  	System.out.println("intId=" + intId);
+    	TeacherLogger tLogger = (TeacherLogger) request.getSession().getAttribute("tLogger");
+    	tLogger.logEntryWorker((int) request.getSession().getAttribute("teacherId"), 0, reportType, classId);
+
+    	System.out.println("TeacherToolsReportController filter = " + filter);
+    	return reportService.generateTeacherReport(teacherId, classId, reportType, lang, filter);
 
     }
 
     @RequestMapping(value = "/tt/getCompleteMasteryProjectionForStudent", method = RequestMethod.POST)
     public @ResponseBody
-    String getCompleteMasteryProjectionForStudent(ModelMap map,@RequestParam("classId") String classId, @RequestParam("chartType") String chartType, @RequestParam("studentId") String studentId) throws TTCustomException {
-    	System.out.println("TeacherToolsReportController");
+    String getCompleteMasteryProjectionForStudent(ModelMap map,@RequestParam("classId") String classId, @RequestParam("chartType") String chartType, @RequestParam("studentId") String studentId, HttpServletRequest request) throws TTCustomException {
+    	System.out.println("getCompleteMasteryProjectionForStudent");
```

Comment:
we should do all the logging after the functionality of that method So that in care of any failure ofthe logging the functionality of the method still stays infact.

Reply:
I put the teacher logging in a try-catch so that it will not impact the code that follows it.